### PR TITLE
sql/stats: Correctly cancel stats background context on quiesce.

### LINKED ISSUE #114296: pkg/ccl/backupccl/backupccl_test_/backupccl_test_shard_21/48: pkg failed
pkg/ccl/backupccl/backupccl_test_/backupccl_test_shard_21/48.pkg [failed](https://tanzanite.cluster.engflow.com/invocation/5851468d-422c-414e-a271-28ffb9d9c8b0) on master @ [cdf92626a841423e6a22fe204475235875883f61](https://github.com/cockroachdb/cockroach/commits/cdf92626a841423e6a22fe204475235875883f61):

```
```
<p>Parameters: <code>attempt=1</code>
, <code>run=12</code>
, <code>shard=21</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pkg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Jira issue: CRDB-33439

### PR DESCRIPTION
**Pull request by benbardin**
Currently, this can get stuck while quiescing in `ensureAllTables()`, where a SQL query is issued and cannot be completed.

Fixes: #114296
Release note: None

### CODE REVIEW

File: pkg/sql/stats/automatic_stats.go [hunk 1]
```diff
@@ -472,8 +473,6 @@ func (r *Refresher) Start(
 							break
 						case <-r.drainAutoStats:
 							return
-						case <-stopper.ShouldQuiesce():
-							return
```
Comment (stevendanna) [inline pkg/sql/stats/automatic_stats.go:476]: If you are going remove the stopper.ShouldQuiesce() I would add a `<-ctx.Done()` branch to ensure we are making an equivalent change. Same in the other two locations.

**Review by stevendanna — CHANGES_REQUESTED** [review]

**Review by stevendanna — APPROVED** [review]
Thanks for looking into this! This change makes sense to me.

**Comment by benbardin** [comment]
TFTR!

**Pull request merged**